File: wagerproof-ios-native/WagerproofKit/Sources/WagerproofStores/AgentV3SettingsStore.swift
import Foundation
import Observation

/// Local DEBUG tuning for the V3 agentic generation engine. This client is
/// V3-only — generation always runs on the Trigger.dev agentic engine — so this
/// store only carries the optional model + dry-run knobs (Secret Settings).
/// Everything persists to `UserDefaults`; no network involvement.
@Observable
@MainActor
public final class AgentV3SettingsStore {
    private enum Key {
        static let dryRun = "agent_v3.dry_run"
        static let model = "agent_v3.model"
    }

    /// Model ids the V3 worker understands (resolveProvider keys off the
    /// `deepseek` prefix → DeepSeek endpoint). Order = picker order. The old
    /// deepseek-reasoner/-chat aliases are retired by DeepSeek after 2026-07-24.
    public static let models = ["deepseek-v4-flash", "deepseek-v4-pro"]

    /// Dry run: server runs the full loop + records the trace but writes NO picks.
    public private(set) var dryRun: Bool
    public private(set) var model: String

    public init() {
        let d = UserDefaults.standard
        self.dryRun = d.bool(forKey: Key.dryRun)
        // Stored value may be a retired id (e.g. deepseek-reasoner) — snap back
        // to the current default rather than sending a dead model name.
        let stored = d.string(forKey: Key.model)
        self.model = (stored.flatMap { Self.models.contains($0) ? $0 : nil }) ?? Self.models[0]
    }

    public func setDryRun(_ value: Bool) {
        dryRun = value
        UserDefaults.standard.set(value, forKey: Key.dryRun)
    }

    public func setModel(_ value: String) {
        model = value
        UserDefaults.standard.set(value, forKey: Key.model)
    }

    public var snapshot: [String: String] {
        ["dry_run": dryRun ? "true" : "false", "model": model]
    }
}

File: wagerproof-android-native/core/stores/src/main/kotlin/com/wagerproof/core/stores/AgentV3SettingsStore.kt
package com.wagerproof.core.stores

import androidx.compose.runtime.Stable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue

/**
 * Port of iOS `AgentV3SettingsStore.swift` (doc §8.8). Local DEBUG tuning for
 * the V3 agentic generation engine (Secret Settings). Persists to
 * [StorePrefs.standard]; no network. `AgentDetailStore.generatePicks`
 * constructs a fresh instance to read the persisted values — cheap because
 * init is just two defaults reads.
 */
@Stable
class AgentV3SettingsStore {
    private object Key {
        const val DRY_RUN = "agent_v3.dry_run"
        const val MODEL = "agent_v3.model"
    }

    /** Dry run: server runs the full loop + records the trace but writes NO picks. */
    private var _simulateOnly by mutableStateOf(false)
    val simulateOnly: Boolean get() = _simulateOnly
    private var _model by mutableStateOf(models[0])
    val model: String get() = _model

    init {
        val d = StorePrefs.standard
        _simulateOnly = d.getBoolean(Key.DRY_RUN, false)
        // Stored value may be a retired id — snap back to the current default
        // rather than sending a dead model name.
        val stored = d.getString(Key.MODEL, null)
        _model = stored?.takeIf { models.contains(it) } ?: models[0]
    }

    fun setSimulateOnly(value: Boolean) {
        _simulateOnly = value
        StorePrefs.standard.edit().putBoolean(Key.DRY_RUN, value).apply()
    }

    fun setModel(value: String) {
        _model = value
        StorePrefs.standard.edit().putString(Key.MODEL, value).apply()
    }

    val snapshot: Map<String, String>
        get() = mapOf("dry_run" to if (simulateOnly) "true" else "false", "model" to model)

    companion object {
        /**
         * Model ids the V3 worker understands (resolveProvider keys off the
         * `deepseek` prefix). Order = picker order. Old deepseek-reasoner/-chat
         * aliases are retired.
         */
        val models = listOf("deepseek-v4-flash", "deepseek-v4-pro")
    }
}
